numpy/lib/stride_tricks.py

```diff
@@ -111,6 +111,133 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
+    a subsets of the array at any window position.
+
+    Parameters
+    ----------
+    x : array_like
+        Array to create the sliding window view from.
+
+    shape : int or tuple of int
+        The shape of the window. If `axis` is not present, must have same
+        length as the number of input array dimensions.
+
+    axis : int or tuple of int, optional
+        If present, `shape[i]` will refer to the axis `axis[i]` of `x`,
+        otherwise `shape[i]` will refer to axis `i` of `x`.
+
+    subok : bool, optional
+        If True, sub-classes will be passed-through, otherwise the returned
+        array will be forced to be a base-class array (default).
+
+    writeable : bool, optional
+        When true, allow writing to the returned view. The default is false,
+        as this should be used with caution: the returned view contains the
+        same memory location multiple times, so writing to one location will
+        cause others to change.
+
+    Returns
+    -------
+    view : ndarray
+        Sliding window view of the array.
+
+    See Also
+    --------
+    as_strided: Create a view into the array with the given shape and strides.
+    broadcast_to: broadcast an array to a given shape.
+
+    Notes
+    -----
+    For some cases there may be more efficient approaches to calculate
+    transformations across multi-dimensional arrays, for instance
+    `scipy.signal.fftconvolve`, where combining the iterating step with the
+    calculation itself while storing partial results can result in significant
+    speedups.
+
+    Examples
+    --------
+    >>> x = np.arange(5)
+    >>> sliding_window_view(x, 2)
+    array([[0, 1],
+           [1, 2],
+           [2, 3],
+           [3, 4]])
+
+    This also works in more dimensions, e.g.
+
+    >>> i, j = np.ogrid[:3, :4]
+    >>> x = 10*i + j
+    >>> x
+    array([[ 0,  1,  2,  3],
+           [10, 11, 12, 13],
+           [20, 21, 22, 23]])
+    >>> shape = (2,2)
+    >>> sliding_window_view(x, shape)
+    array([[[[ 0,  1],
+             [10, 11]],
+            [[ 1,  2],
+             [11, 12]],
+            [[ 2,  3],
+             [12, 13]]],
+           [[[10, 11],
+             [20, 21]],
+            [[11, 12],
+             [21, 22]],
+            [[12, 13],
+             [22, 23]]]])
+
+    The axis can be specified explicitly:
+
+    >>> sliding_window_view(x, 3, 0)
+    array([[[ 0, 10, 20],
+            [ 1, 11, 21],
+            [ 2, 12, 22],
+            [ 3, 13, 23]]])
+
+    The same axis can be used several times:
+
+    >>> sliding_window_view(x, (2, 3), (1, 1))
+    array([[[[ 0,  1,  2],
+             [ 1,  2,  3]]],
+           [[[10, 11, 12],
+             [11, 12, 13]]],
+           [[[20, 21, 22],
+             [21, 22, 23]]]])
+    """
+    shape = tuple(shape) if np.iterable(shape) else (shape,)
+    # first convert input to array, possibly keeping subclass
+    x = np.array(x, copy=False, subok=subok)
```

Comment:
```suggestion
    x = np.asanyarray(x)
```

Reply:
This was briefly discussed in the previous PR https://github.com/numpy/numpy/pull/10771#discussion_r317427162. There, @eric-wieser suggested that this was done on purpose and to conform with `as_strided`. That's why I kept it, but I have to admit that I don't understand the issue in detail.

@eric-wieser, would you care to way in?

Reply:
IMO this sounds more like an argument for changing `as_strided`. 
Still, as this is quite minor I don't have particularly strong feelings about either option; feel free to ignore.

Reply:
I think we should leave this as is for now. `_broadcast_to` also uses the same pattern. @BvB93, if you feel strongly about this feel welcome to open a follow-up PR that changes all the conversions in this file.